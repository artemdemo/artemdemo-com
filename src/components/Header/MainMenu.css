.main-menu-link:hover {
    text-decoration: none;
}