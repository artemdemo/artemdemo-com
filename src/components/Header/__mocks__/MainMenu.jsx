import React from 'react';

const MainMenu = props => (
    <div data-mock='MainMenu'>
        {JSON.stringify(props, null, 2)}
    </div>
);

export default MainMenu;
