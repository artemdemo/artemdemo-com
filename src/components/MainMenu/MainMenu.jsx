import React from 'react';
import { Link } from 'gatsby';

const MainMenu = () => {
    return (
        <ul>
            <li>
                <Link
                    to='/about'
                >
                    About me
                </Link>
            </li>
            <li>
                <Link
                    to='/contact'
                >
                    Contact me
                </Link>
            </li>
        </ul>
    );
};

export default MainMenu;