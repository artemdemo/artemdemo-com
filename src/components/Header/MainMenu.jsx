import React from 'react';
import styled from 'styled-components';
import { Link } from 'gatsby';
import { locationProp } from '../Layout/layoutProps';
import * as styles from '../../services/styles';

import './MainMenu.css';

const MainMenuSty = styled.ul`
    ${styles.clearFix}
    ${styles.resetUl}
`;

const MainMenuItemSty = styled.li`
    float: left;
    padding: 0 10px;
    ${props => props.active && `
        border-bottom: 2px solid #96c9ff;
    `}
`;

const MainMenu = (props) => {
    const { location } = props;
    return (
        <MainMenuSty>
            <MainMenuItemSty active={location.pathname === '/' || location.pathname.startsWith('/blog')}>
                <Link className='main-menu-link' to='/'>
                    Blog
                </Link>
            </MainMenuItemSty>
            <MainMenuItemSty active={location.pathname.startsWith('/about')}>
                <Link className='main-menu-link' to='/about'>
                    About
                </Link>
            </MainMenuItemSty>
            <MainMenuItemSty active={location.pathname.startsWith('/contact')}>
                <Link className='main-menu-link' to='/contact'>
                    Contact me
                </Link>
            </MainMenuItemSty>
        </MainMenuSty>
    );
};

MainMenu.propTypes = {
    location: locationProp,
};

MainMenu.defaultProps = {
    location: {},
};

export default MainMenu;