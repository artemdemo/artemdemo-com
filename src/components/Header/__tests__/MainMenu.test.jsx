import React from 'react';
import renderer from 'react-test-renderer';
import MainMenu from '../MainMenu';

describe('MainMenu', () => {
    it('should render without location', () => {
        const tree = renderer.create(
            <MainMenu />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with pathname = `/`', () => {
        const tree = renderer.create(
            <MainMenu
                location={{pathname: '/'}}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with pathname = `/blog`', () => {
        const tree = renderer.create(
            <MainMenu
                location={{pathname: '/blog'}}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with pathname = `/about`', () => {
        const tree = renderer.create(
            <MainMenu
                location={{pathname: '/about'}}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with pathname = `/contact`', () => {
        const tree = renderer.create(
            <MainMenu
                location={{pathname: '/contact'}}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
