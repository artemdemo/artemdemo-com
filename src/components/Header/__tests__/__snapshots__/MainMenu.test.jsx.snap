// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu should render with pathname = \`/\` 1`] = `
<ul
  className="MainMenu__MainMenuSty-sc-10g5x94-0 frjzoV"
>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 hjxubK"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Blog
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/about",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      About
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/contact",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Contact me
    </div>
  </li>
</ul>
`;

exports[`MainMenu should render with pathname = \`/about\` 1`] = `
<ul
  className="MainMenu__MainMenuSty-sc-10g5x94-0 frjzoV"
>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Blog
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 hjxubK"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/about",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      About
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/contact",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Contact me
    </div>
  </li>
</ul>
`;

exports[`MainMenu should render with pathname = \`/blog\` 1`] = `
<ul
  className="MainMenu__MainMenuSty-sc-10g5x94-0 frjzoV"
>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 hjxubK"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Blog
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/about",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      About
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/contact",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Contact me
    </div>
  </li>
</ul>
`;

exports[`MainMenu should render with pathname = \`/contact\` 1`] = `
<ul
  className="MainMenu__MainMenuSty-sc-10g5x94-0 frjzoV"
>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Blog
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/about",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      About
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 hjxubK"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/contact",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Contact me
    </div>
  </li>
</ul>
`;

exports[`MainMenu should render without location 1`] = `
<ul
  className="MainMenu__MainMenuSty-sc-10g5x94-0 frjzoV"
>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Blog
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/about",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      About
    </div>
  </li>
  <li
    className="MainMenu__MainMenuItemSty-sc-10g5x94-1 iErzFj"
  >
    <div
      data-mock="Link"
    >
      {
  "to": "/contact",
  "className": "MainMenu__MainMenuLinkSty-sc-10g5x94-2 edNkKg"
}
      Contact me
    </div>
  </li>
</ul>
`;
